ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -444,11 +446,17 @@ private[dao] final class TransactionsReader(
         )
       }
       .map(event => (event.eventOffset, event.eventSequentialId) -> event)
-      .mapMaterializedValue(_ => NotUsed)
+
+    val groupedByOffset = TransactionsReader
+      .groupContiguous(contractStateEventsSource)(by = { case ((offset, _), _) => offset })
```

Comment:
Other streams are grouped by `transactionId`. Does anyone see any problem of grouping here by `offset` instead?